daphne/src/roles/helper.rs

```diff
@@ -140,22 +147,71 @@ pub trait DapHelper<S>: DapAggregator<S> {
                         "unexpected message for aggregation job (already exists)".into(),
                     ));
                 }
+                metrics.agg_job_started_inc();
                 agg_job_resp
             }
-            DapHelperTransition::Finish(..) => {
-                return Err(fatal_error!(err = "unexpected transition (finished)").into());
+            DapHelperAggregationJobTransition::Finished(
+                mut maybe_agg_share_span,
+                mut agg_job_resp,
+            ) => {
+                // XXX This loop is messier than I'd like, but should be cleaned up after
+                // https://github.com/cloudflare/daphne/issues/408.
```

Comment:
Reviewer note: Before cleaning this up, I'd like to see where we land on this #408.